### PR TITLE
ProjectOptions (public to private) & Project struct (org related fields) fixes 

### PR DESCRIPTION
**Pull request by DariuszPorowski**
This fixes https://github.com/google/go-github/issues/2110

- fixed ProjectOptions: Public -> Private (according to API doc: https://docs.github.com/en/rest/reference/projects#update-a-project)
- added **OrganizationPermission** and **Private** fields to Project struct, which presents during response on Organization Project.

I've done UATs as well, and all API calls looked good:
- Update Org Project with new private field
- Get Org Project (OrganizationPermission and Private presents as expected)
- Get Repo Project (OrganizationPermission and Private not presents as expected)
- Get User Project (OrganizationPermission and Private not presents as expected)

### CODE REVIEW

**Comment by DariuszPorowski** [comment]
@googlebot I fixed it.

**Review by gmlewis — APPROVED** [review]
Thank you, @DariuszPorowski !
LGTM.

Awaiting second LGTM before merging.

This is technically a Breaking API change... however, I'm tempted to declare this one as non-breaking simply because what we are changing was already broken... so in theory, any attempts to use the prior version would have failed.

Thoughts?  Is it OK to _NOT_ declare this PR as a breaking APi change (and therefore not have to bump the major version number of this repo at the next release due to this PR alone)?

@willnorris - what is your opinion?

**Comment by willnorris** [comment]
Even though this was never working as intended, it's a breaking change in the Go API in that if anyone *was* calling it, their code would no longer compile now.  And since it's been in the wild [for 3 years](https://github.com/google/go-github/commit/d8bee7566740e39c1c0e2c3990e1e5f348458852), I feel like it would be more responsible to treat it as a breaking change.

**Comment by DariuszPorowski** [comment]
@gmlewis I agree with @willnorris
Even if it is a change to a valid GitHub API, it changes the entire usage of Go API.

**Review by cpanato — APPROVED** [review]
lgtm
we need to document that in the release notes

**Comment by gmlewis** [comment]
Thank you, @cpanato - merging.

**Pull request merged**
